Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/QuotaCreatedForTierIntegrationEventHandlerTests.cs

```diff
@@ -43,9 +45,38 @@ public async void Creates_tier_quota_after_consuming_integration_event()
         ).MustHaveHappened();
     }
 
+    [Fact]
+    public async void Updates_metric_statuses_after_creating_tier_quota()
+    {
+        // Arrange
+        var tierId = new TierId("TIRFxoL0U24aUqZDSAWc");
+        var tier = new Tier(tierId, "some-tier-name");
+
+        var max = 5;
+        var period = QuotaPeriod.Month;
+        var tierQuotaDefinition = new TierQuotaDefinition(MetricKey.NumberOfSentMessages, max, period);
+        var tierQuotaDefinitionsRepository = new FindTierQuotaDefinitionsStubRepository(tierQuotaDefinition);
+
+        var firstIdentity = new Identity("some-identity-address-one", tierId);
+        var secondIdentity = new Identity("some-identity-address-two", tierId);
+        var identities = new List<Identity> { firstIdentity, secondIdentity };
+        var identitiesRepository = A.Fake<IIdentitiesRepository>();
+        A.CallTo(() => identitiesRepository.FindWithTier(tierId, CancellationToken.None, true)).Returns(identities);
+
+        var handler = CreateHandler(identitiesRepository, tierQuotaDefinitionsRepository);
+
+        // Act
+        await handler.Handle(new QuotaCreatedForTierIntegrationEvent(tier.Id, tierQuotaDefinition.Id));
+
+        // Assert
+        A.CallTo(() => _metricStatusesService.RecalculateMetricStatuses(A<List<string>>.That.Matches(x => x.Contains("some-identity-address-one") && x.Contains("some-identity-address-two")),
+            A<List<string>>.That.Contains(tierQuotaDefinition.MetricKey.Value), A<CancellationToken>._)).MustHaveHappened();
```

Comment:
What if "some-identity-address-three" was passed as well? (=> check for the length)